Bottom: 957d9f6b42aa6f3a71089907db78bb1f00e954eb
Top:    9b85afcf26d3db7711c4d9254af03f2727a7e4b8
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-23 16:56:13 +0100

draw.c: Change DrawTop() to DrawOuter() taking a side parameter


---

diff --git a/src/draw.c b/src/draw.c
index 88c90cb..f67aa29 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -83,7 +83,7 @@ static const BoxType *clip_box = NULL;
  */
 static void Redraw (bool, BoxTypePtr);
 static void DrawEverything (BoxTypePtr);
-static void DrawTop (const BoxType *);
+static void DrawOuterLayer (int side, const BoxType *);
 static int DrawLayerGroup (int, const BoxType *);
 static void DrawPinOrViaLowLevel (PinTypePtr, bool);
 static void DrawPlainPin (PinTypePtr, bool);
@@ -315,7 +315,7 @@ PrintAssembly (const BoxType * drawn_area, int side_group, int swap_ident)
   gui->set_draw_faded (Output.fgGC, 1);
   SWAP_IDENT = swap_ident;
   DrawLayerGroup (side_group, drawn_area);
-  DrawTop (drawn_area);
+  DrawOuterLayer (side, drawn_area);
   gui->set_draw_faded (Output.fgGC, 0);
 
   /* draw package */
@@ -414,9 +414,11 @@ DrawEverything (BoxTypePtr drawn_area)
   if (TEST_FLAG (CHECKPLANESFLAG, PCB) && gui->gui)
     return;
 
+  side = SWAP_IDENT ? SOLDER_LAYER : COMPONENT_LAYER;
+
   /* Draw pins, pads, vias below silk */
   if (gui->gui)
-    DrawTop (drawn_area);
+    DrawOuterLayer (side, drawn_area);
   else
     {
       HoleCountStruct hcs;
@@ -581,10 +583,8 @@ pad_callback (const BoxType * b, void *cl)
  * draws pins pads and vias
  */
 static void
-DrawTop (const BoxType * screen)
+DrawOuterLayer (side, const BoxType * screen)
 {
-  int side = SWAP_IDENT ? SOLDER_LAYER : COMPONENT_LAYER;
-
   if (PCB->PinOn || doing_assy)
     {
       /* draw element pins */
